blockchain/pool.go

```diff
@@ -283,34 +278,22 @@ func (pool *BlockPool) decrPeer(peerId string) {
 	peer.numRequests--
 }
 
-func (pool *BlockPool) nextHeight() uint {
-	pool.requestsMtx.Lock() // Lock
-	defer pool.requestsMtx.Unlock()
-
-	// we make one request per height.
-	return pool.height + uint(pool.numTotal)
-}
-
-func (pool *BlockPool) makeRequest(height uint) {
+func (pool *BlockPool) makeNextRequest() {
 	pool.requestsMtx.Lock() // Lock
 	defer pool.requestsMtx.Unlock()
 
+	nextHeight := pool.height + uint(len(pool.requests)) + 1
```

Comment:
Don't think we need the +1. pool.height starts at 1 and the requests are empty so this will actually skip the first block and cause block pool to hang. we get the +1 by adding the new request to the requests map on 292